rfc/rfc-0020-resources-update-after-success.md

```diff
@@ -0,0 +1,91 @@
+# Draft RFC 20 Update Resources Only After Success/Failure
+
+## Summary
+
+Cartographer is currently unable to determine when an input to a resource has reached success, failed, or is still
+processing. Making Carto aware of status indications is not sufficient, as frequent updates to a resource could keep it
+in a constant state of 'processing', even as it succeeds and fails on multiple successive inputs. Carto should wait
+until an input has resulted in success or failure before updating the resource with new input. 
+
+## Motivation
+
+Connecting an output of a resource to an input is necessary for establishing provenance. That is to state, "The app 
+currently running on the cluster is a result of commit X," it is necessary to tie a resource output to the input that
+produced it. Waiting for success/failure before update achieves this.
```

Comment:
> The app currently running on the cluster is a result of commit X

I don't believe this is reasonably knowable as there are many triggers that can result in a templated component updating it's output. Taking just a component like kpack, a new image could be produced because of a source change, or a buildpack update or a run image update.

Suppose I have a supply chain that runs unit tests on the source code. If a new commit is detected and the unit tests fail. Should the whole supply chain be haunted, or should downstream components continue to operate with the last known good values?

Reply:
1. I don't mean to starve the supply chain of the last known good value of a resource. I've added a section detailing a strategy forward on that. [Link](https://github.com/vmware-tanzu/cartographer/blob/rfc-0020-template-success/rfc/rfc-0020-resources-update-after-success.md#deadlock)
2. "I don't believe this is reasonably knowable" I am glad that you are pushing on this point. I _think_ that this is knowable but have not yet completely convinced myself. I expect this RFC to either generate a counter-example that proves our hubris or to give us confidence through the stress testing of many eyes. Gaming out scenarios just now pushed me to a similar (but slightly different strategy). I've written up that thinking and scenarios. https://github.com/vmware-tanzu/cartographer/blob/rfc-0020-template-success/rfc/rfc-0020-resources-update-after-success.md#proposed-solution I believe this covers the scenario you posited above. I'm very interested in hearing about other corner cases!